routing/missioncontrol_store.go

```diff
@@ -359,68 +359,98 @@ func (b *missionControlStore) storeResults() error {
 	b.queueCond.L.Unlock()
 
 	var (
-		keys       *list.List
-		keysMap    map[string]struct{}
+		newKeys    map[string]struct{}
+		delKeys    []string
 		storeCount int
 		pruneCount int
 	)
 
+	// Create a deduped list of new entries.
+	newKeys = make(map[string]struct{}, l.Len())
+	for e := l.Front(); e != nil; e = e.Next() {
+		pr, ok := e.Value.(*paymentResult)
+		if !ok {
+			panic(fmt.Sprintf("wrong type %T (not *paymentResult)",
```

Comment:
please dont use `panic` 🙏  return an error rather

Reply:
Just to give my reasoning, I thought it would be ok in this situation because this is really a programmer error (no different than not checking for a nil argument on every function) and this was entirely avoidable by using the generic version of the lists (which was deemed as unnecessary in the review).

But in any case, I'll address it next.